### PR TITLE
Simplify kv import

### LINKED ISSUE #4103: KV simplify import implementation

### PR DESCRIPTION
**Pull request by nopcoder**
- Reader not as part of go routine
- Reader check for returned error and return

Close #4103

### CODE REVIEW

File: pkg/kv/import.go [hunk 1]
```diff
@@ -127,16 +103,35 @@ func Import(ctx context.Context, reader io.Reader, store Store) error {
...
+	err := importReader(ctx, log, entryChan, jd)
+	close(entryChan)
+
+	wg.Wait()
 
-	return g.Wait().ErrorOrNil()
+	var merr *multierror.Error
+	merr = multierror.Append(merr, err)
+	merr = multierror.Append(merr, workerErr...)
+	return merr.ErrorOrNil()
```
Comment (N-o-Z) [inline pkg/kv/import.go:136]: multiErr better?

File: pkg/kv/import.go [hunk 2]
```diff
@@ -127,16 +103,35 @@ func Import(ctx context.Context, reader io.Reader, store Store) error {
 	})
 	log.Info("Processing file")
 
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
```
Comment (N-o-Z) [inline pkg/kv/import.go:107]: Why do we need this?

**Review by N-o-Z — APPROVED** [review]
Looks great!
1 question, 1 comment

File: pkg/kv/import.go (hunk 1 above)
Comment (nopcoder) [reply pkg/kv/import.go:136]: This is the same type used by `m.Go` - we can revert part of my change to use it and simplify it a bit more

File: pkg/kv/import.go (hunk 2 above)
Comment (nopcoder) [reply pkg/kv/import.go:107]: prevent resource leak

File: pkg/kv/import.go (hunk 1 above)
Comment (N-o-Z) [reply pkg/kv/import.go:136]: I was referring to the variable name

File: pkg/kv/import.go (hunk 2 above)
Comment (nopcoder) [reply pkg/kv/import.go:103]: https://pkg.go.dev/context#WithCancel
with the example as good resource

File: pkg/kv/import.go [hunk 3]
```diff
@@ -127,16 +99,27 @@ func Import(ctx context.Context, reader io.Reader, store Store) error {
...
 		})
 	}
-	g.Go(func() error {
-		return producer(cctx, cancel, log, entryChan, jd)
-	})
+	err := importReader(ctx, log, entryChan, jd)
+	close(entryChan)
 
-	return g.Wait().ErrorOrNil()
+	merr := g.Wait()
```
Comment (N-o-Z) [inline pkg/kv/import.go:122]: Can you just change the variable name

**Review by N-o-Z — APPROVED** [review]
Thanks

File: pkg/kv/import.go (hunk 3 above)
Comment (nopcoder) [reply pkg/kv/import.go]: Updated

**Pull request merged**